### ISSUE TITLE
Disable @CheckReturnValue checking if the next statement is "fail();"

### ISSUE THREAD

**Issue by cpovirk** [google/error-prone#385]
In adopting `@CheckReturnValue`, I'm noticing a common pattern in tests:

```
try {
  Foo.newFoo(-1);
  fail();
} catch (IllegalArgumentException expected) {
}
```
I end up changing it to:
```
try {
  Foo unused = Foo.newFoo(-1);
  fail();
} catch (IllegalArgumentException expected) {
}
```
That works fine. But I'm wondering if Error Prone could notice this particular pattern and let it pass. Are there any problems it would miss as a result?

**Comment by cushon** [comment]
@kluever reported this internally, it's on my list. I think it'd be fine to relax the check when the next statement is `fail();` and there's an enclosing try/catch.

**Comment by cushon** [comment]
Fixed in d3342905780eca7eeec4c0fcdbd8b8c5d5acb245
